### PR TITLE
Implements `ipinfo tool is_valid`

### PR DESCRIPTION
**Pull request by Harisabdullah**

### CODE REVIEW

File: ipinfo/cmd_tool.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ Commands:
   prev         get the previous IP of the input IP
   is_v4        reports whether input is an IPv4 address.
   is_v6        reports whether input is an IPv6 address.
+  isValid      reports whether the Addr is an initialized address (not the zero Addr).
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:42]: We can't be using internal technical jargon in the user-facing help message

```suggestion
  isValid      reports whether an IP is valid.
```

File: ipinfo/cmd_tool.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ var completionsTool = &complete.Command{
 		"next":      completionsToolNext,
 		"prev":      completionsToolPrev,
 		"is_v4":     completionsToolIsV4,
+		"isValid":   completionsToolIsValid,
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:18]: Should be `is_valid`.

File: ipinfo/cmd_tool_isValid.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	},
+}
+
+// printHelpToolIsValid prints the help message for the "IsValid" command.
+func printHelpToolIsValid() {
+	fmt.Printf(
+		`Usage: %s tool IsValid <ip>
+
+Description:
+  reports whether the Addr is an initialized address (not the zero Addr).
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isValid.go:24]: Same as above

File: ipinfo/cmd_tool_isValid.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Examples:
+  %[1]s IsValid "190.87.89.1"
+  %[1]s IsValid "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
+  %[1]s IsValid "::"
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isValid.go:29]: Do provide an invalid example if not already

File: lib/cmd_tool_valid.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		printHelp()
+		return nil
+	}
+
+	op := func(input string, input_type INPUT_TYPE) error {
+		switch input_type {
+		case INPUT_TYPE_IP:
+			fmt.Printf("%s %v\n", input, true)
+		default:
+			fmt.Printf("%s %v\n", input, false)
```
Comment (UmanShahzad) [inline lib/cmd_tool_valid.go:35]:
```suggestion
			fmt.Printf("%s,%v\n", input, true)
		default:
			fmt.Printf("%s,%v\n", input, false)
```

**Review by UmanShahzad — COMMENTED** [review]
Needs to follow subcmd naming style as used in others (snake case not pascal / camel case).

File: ipinfo/cmd_tool_isValid.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	},
+}
+
+// printHelpToolIsValid prints the help message for the "is_valid" command.
+func printHelpToolIsValid() {
+	fmt.Printf(
+		`Usage: %s tool is_valid <ip>
+
+Description:
+  reports whether the ip address is an initialized address (not the zero Addr).
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isValid.go:24]: I left a comment here as well to update this desc to match the short desc above

**Review by UmanShahzad — APPROVED** [review]
One small fix

File: ipinfo/cmd_tool_isValid.go (hunk 6 above)
Comment (Harisabdullah) [reply ipinfo/cmd_tool_isValid.go:24]: Sure! I'll take care of it

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**